src/Ccap/Codegen/Annotations.purs

```diff
@@ -10,11 +13,14 @@ import Control.Alt ((<|>))
 import Data.Array as Array
 import Data.Maybe (Maybe)
 
-field :: String -> String -> Annotations -> Maybe String
-field annotKey paramKey annots = do
+fieldWithOptParamValue :: String -> String -> Annotations -> Maybe (Maybe String)
+fieldWithOptParamValue annotKey paramKey annots = do
   Annotation _ _ params <- Array.find (\(Annotation n _ _) -> n == annotKey) annots
   AnnotationParam _ _ v <- Array.find (\(AnnotationParam n _ _) -> n == paramKey) params
-  v
+  pure v
+
+field :: String -> String -> Annotations -> Maybe String
+field annotKey paramKey annots = join $ fieldWithOptParamValue annotKey paramKey annots
```

Comment:
`field = join <<< fieldWithOptParamValue`, if you're into the whole point-free thing.

Reply:
unfortunately that wont quite work since `fieldWithOptParamValue` is type `a -> b -> c -> d` and join is `d -> e`. I can make it 
```field annotKey paramKey = join <<< fieldWithOptParamValue annotKey paramKey``` 
though